# Notebook 01 - Veri Keşfi ve Tanımlayıcı İstatistik

## Enerji Perakende Veri Analizi Case Study

### Amaç

Bu notebook, Ahmet Çalık Vakfı İleri Veri Analitiği Eğitimi kapsamında verilen
Enerji Perakende Veri Analizi Case Study çalışmasının ilk bölümünü içermektedir.

Bu notebook içerisinde;

- Veri seti yüklenmektedir.
- Veri yapısı incelenmektedir.
- Tanımlayıcı istatistikler hesaplanmaktadır.
- Veri kalitesi kontrol edilmektedir.
- İlçeler bazında müşteri sayıları analiz edilmektedir.
- Elektrik tüketim verileri incelenmektedir.

Bu notebook yalnızca veri keşfi ve tanımlayıcı istatistik çalışmalarını kapsamaktadır.

## 1. Veri Setinin Yüklenmesi

Bu bölümde gerekli Python kütüphaneleri içe aktarılmakta, Excel dosyası okunmakta, sayfa isimleri kontrol edilmekte ve her sayfa ayrı bir DataFrame olarak yüklenmektedir.


In [1]:
import pandas as pd
import numpy as np

pd.set_option("display.max_columns", None)
pd.set_option("display.max_rows", 20)

In [ ]:
# Excel dosyasının yolu
file_path = "../data/elektrik_veri.xlsx"

# Excel dosyasını açma
xls = pd.ExcelFile(file_path)

# Sayfa isimlerini görüntüleme
print("Sayfa isimleri:")
print(xls.sheet_names)

Sayfa isimleri:
['Tahsilat', 'Tahsilat 1', 'Tahakkuk', 'Tahakkuk 1', 'Tahakkuk 2']


In [ ]:
# Her sayfayı ayrı bir DataFrame olarak yükleme

df_tahsilat = pd.read_excel(xls, sheet_name="Tahsilat")

df_tahsilat_1 = pd.read_excel(xls, sheet_name="Tahsilat 1")

df_tahakkuk = pd.read_excel(xls, sheet_name="Tahakkuk")

df_tahakkuk_1 = pd.read_excel(xls, sheet_name="Tahakkuk 1")

df_tahakkuk_2 = pd.read_excel(xls, sheet_name="Tahakkuk 2")

---
## 2. Veri Setlerinin Boyutlarının İncelenmesi

Bu bölümde her veri setinin satır ve sütun sayıları kontrol edilmektedir.

Bu kontrol sayesinde veri setlerinin eksiksiz yüklendiği ve proje dokümanındaki referans değerlerle uyumlu olduğu doğrulanmaktadır.


In [4]:
print("Tahsilat      :", df_tahsilat.shape)
print("Tahsilat 1    :", df_tahsilat_1.shape)
print("Tahakkuk      :", df_tahakkuk.shape)
print("Tahakkuk 1    :", df_tahakkuk_1.shape)
print("Tahakkuk 2    :", df_tahakkuk_2.shape)

Tahsilat      : (636993, 9)
Tahsilat 1    : (917632, 22)
Tahakkuk      : (124818, 10)
Tahakkuk 1    : (765657, 10)
Tahakkuk 2    : (295223, 10)


---
## 3. Veri Yapısının İncelenmesi

Bu bölümde veri setlerinin yapısı incelenmektedir.

Her veri seti için;

- sütun isimleri,
- veri tipleri,
- eksik değer durumu,
- ilk beş kayıt

incelenerek veri yapısı doğrulanmaktadır.


In [5]:
# Tüm veri setlerinin bilgi çıktıları

dataframes = {
    "Tahsilat": df_tahsilat,
    "Tahsilat 1": df_tahsilat_1,
    "Tahakkuk": df_tahakkuk,
    "Tahakkuk 1": df_tahakkuk_1,
    "Tahakkuk 2": df_tahakkuk_2
}

for isim, df in dataframes.items():
    print("=" * 70)
    print(f"{isim} Veri Seti")
    print("=" * 70)
    df.info()
    print("\n")

Tahsilat Veri Seti
<class 'pandas.DataFrame'>
RangeIndex: 636993 entries, 0 to 636992
Data columns (total 9 columns):
 #   Column                 Non-Null Count   Dtype         
---  ------                 --------------   -----         
 0   Şube                   636993 non-null  str           
 1   Kasa                   636993 non-null  str           
 2   İlçe                   636993 non-null  str           
 3   Söz.hsp.(bağımsız)     636993 non-null  int64         
 4   Tahsilat Tarihi        636993 non-null  datetime64[us]
 5   Nakit Tahsilat         523 non-null     float64       
 6   Mahsuben Tahsilat      7542 non-null    float64       
 7   Kredi Kartı Tahsilatı  0 non-null       float64       
 8   Banka Tahsilatı        628933 non-null  float64       
dtypes: datetime64[us](1), float64(4), int64(1), str(3)
memory usage: 43.7 MB


Tahsilat 1 Veri Seti
<class 'pandas.DataFrame'>
RangeIndex: 917632 entries, 0 to 917631
Data columns (total 22 columns):
 #   Column          

In [ ]:
# İlk 5 satırı görüntüleme

for isim, df in dataframes.items():
    print("=" * 80)
    print(f"{isim} Veri Seti - İlk 5 Satır")
    print("=" * 80)
    display(df.head())

Tahsilat Veri Seti - İlk 5 Satır


,Şube,Kasa,İlçe,Söz.hsp.(bağımsız),Tahsilat Tarihi,Nakit Tahsilat,Mahsuben Tahsilat,Kredi Kartı Tahsilatı,Banka Tahsilatı
0,Tayin edilmedi,Tayin edilmedi,TAŞOVA,4989745446,2023-11-06,NaN,8648.95,NaN,NaN
1,Tayin edilmedi,Tayin edilmedi,TAŞOVA,4989745446,2024-06-26,NaN,762.40,NaN,NaN
2,Tayin edilmedi,Tayin edilmedi,TAŞOVA,4989745446,2024-07-10,NaN,311.60,NaN,NaN
3,PTT,PTT/PV,TAŞOVA,4254955886,2023-01-19,NaN,NaN,NaN,130.5
4,PTT,PTT/PV,TAŞOVA,4254955886,2023-02-17,NaN,NaN,NaN,117.0


Tahsilat 1 Veri Seti - İlk 5 Satır


,Mali yıl/dönem,İl,İlçe,Söz.hsp.(bağımsız),Hesap Sınıfı,Tahakkuk Tutar,Son Ödeme Tarihinden Önceki Tahsilat,Son Ödeme Tarihindeki Tahsilat,Son Ödeme (1),Son Ödeme (2),Son Ödeme (3),Son Ödeme (4),Son Ödeme (5),Son Ödeme (6-10),Son Ödeme (10-20),Son Ödeme (20-30),Son Ödeme (30-60),Son Ödeme (60-90),Son Ödeme (90-120),Son Ödeme (120-150),Son Ödeme (150-180),Son Ödeme (180+)
0,OCK 2023,AMASYA,GÖYNÜCEK,9374624783,Mesken,5.03,0.03,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,5.0,NaN,NaN,NaN
1,OCK 2023,AMASYA,GÖYNÜCEK,236184905,Mesken,26.46,0.06,26.40,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,OCK 2023,AMASYA,GÖYNÜCEK,9657731015,Mesken,121.53,NaN,NaN,NaN,121.53,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,OCK 2023,AMASYA,GÖYNÜCEK,9554442880,Mesken,117.49,NaN,117.49,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,OCK 2023,AMASYA,GÖYNÜCEK,6031642522,Mesken,170.30,170.30,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


Tahakkuk Veri Seti - İlk 5 Satır


,il,ilce,sozlesme_hesap_no,mali_yil_donem,fatura_tarihi,kayit_tarihi,vade_tarihi,hesap_sinifi,Hesap Sınıfı,kwh
0,AMASYA,HAMAMÖZÜ,917576806,2023-01-01,2023-01-12,2023-03-06,2023-01-23,M001,Mesken,1.79
1,AMASYA,HAMAMÖZÜ,917576806,2023-01-01,2023-02-09,2023-05-11,2023-02-20,M001,Mesken,2.60
2,AMASYA,HAMAMÖZÜ,917576806,2023-02-01,2023-02-09,2023-05-11,2023-02-20,M001,Mesken,1.23
3,AMASYA,HAMAMÖZÜ,917576806,2023-02-01,2023-03-10,2023-05-11,2023-03-20,M001,Mesken,2.56
4,AMASYA,HAMAMÖZÜ,917576806,2023-03-01,2023-03-10,2023-05-11,2023-03-20,M001,Mesken,1.35


Tahakkuk 1 Veri Seti - İlk 5 Satır


,il,ilce,sozlesme_hesap_no,mali_yil_donem,fatura_tarihi,kayit_tarihi,vade_tarihi,hesap_sinifi,Hesap Sınıfı,kwh
0,AMASYA,GÜMÜŞHACIKÖY,7444449517,2023-01-01,2023-01-11,2023-03-06,2023-01-23,M001,Mesken,21.85
1,AMASYA,GÜMÜŞHACIKÖY,7444449517,2023-01-01,2023-02-10,2023-05-11,2023-02-20,M001,Mesken,44.50
2,AMASYA,GÜMÜŞHACIKÖY,7444449517,2023-02-01,2023-02-10,2023-05-11,2023-02-20,M001,Mesken,22.25
3,AMASYA,GÜMÜŞHACIKÖY,7444449517,2023-02-01,2023-03-10,2023-05-11,2023-03-20,M001,Mesken,45.71
4,AMASYA,GÜMÜŞHACIKÖY,7444449517,2023-03-01,2023-03-10,2023-05-11,2023-03-20,M001,Mesken,25.40


Tahakkuk 2 Veri Seti - İlk 5 Satır


,il,ilce,sozlesme_hesap_no,mali_yil_donem,fatura_tarihi,kayit_tarihi,vade_tarihi,hesap_sinifi,Hesap Sınıfı,kwh
0,AMASYA,GÖYNÜCEK,9374624783,2023-01-01,2023-01-14,2023-03-06,2023-01-24,M001,Mesken,0.10
1,AMASYA,GÖYNÜCEK,9374624783,2023-01-01,2025-03-12,2025-05-09,2025-03-24,M001,Mesken,0.12
2,AMASYA,GÖYNÜCEK,9374624783,2023-02-01,2025-03-12,2025-05-09,2025-03-24,M001,Mesken,0.20
3,AMASYA,GÖYNÜCEK,9374624783,2023-03-01,2025-03-12,2025-05-09,2025-03-24,M001,Mesken,0.00
4,AMASYA,GÖYNÜCEK,9374624783,2023-04-01,2025-03-12,2025-05-09,2025-03-24,M001,Mesken,0.08


---
## 4. Tanımlayıcı İstatistikler

Bu bölümde veri setlerinin temel istatistiksel özellikleri incelenmektedir.

Ortalama, minimum, maksimum, medyan, standart sapma ve çeyrek değerleri değerlendirilerek veri yapısı hakkında genel bir fikir edinilmektedir.

In [7]:
# Tahsilat veri seti

df_tahsilat.describe()

,Söz.hsp.(bağımsız),Tahsilat Tarihi,Nakit Tahsilat,Mahsuben Tahsilat,Kredi Kartı Tahsilatı,Banka Tahsilatı
count,6.369930e+05,636993,523.000000,7542.000000,0.0,628933.000000
mean,5.019884e+09,2024-03-05 09:26:07.644856,694.966635,6180.182282,NaN,372.629109
min,1.758320e+05,2023-01-01 00:00:00,7.450000,-34508.950000,NaN,0.010000
25%,2.532887e+09,2023-07-28 00:00:00,425.330000,44.477500,NaN,120.000000
50%,5.008502e+09,2024-02-26 00:00:00,524.670000,290.410000,NaN,208.000000
75%,7.525722e+09,2024-09-30 00:00:00,688.830000,2729.110000,NaN,322.000000
max,9.999600e+09,2025-05-31 00:00:00,11373.740000,399526.780000,NaN,606473.800000
std,2.884435e+09,NaN,758.319428,23828.022593,NaN,3265.430202


In [8]:
# Diğer veri setlerinin tanımlayıcı istatistikleri

print("Tahsilat 1 Veri Seti")
display(df_tahsilat_1.describe())

print("Tahakkuk - Hamamözü")
display(df_tahakkuk.describe())

print("Tahakkuk 1 - Gümüşhacıköy")
display(df_tahakkuk_1.describe())

print("Tahakkuk 2 - Göynücek")
display(df_tahakkuk_2.describe())

Tahsilat 1 Veri Seti


,Söz.hsp.(bağımsız),Tahakkuk Tutar,Son Ödeme Tarihinden Önceki Tahsilat,Son Ödeme Tarihindeki Tahsilat,Son Ödeme (1),Son Ödeme (2),Son Ödeme (3),Son Ödeme (4),Son Ödeme (5),Son Ödeme (6-10),Son Ödeme (10-20),Son Ödeme (20-30),Son Ödeme (30-60),Son Ödeme (60-90),Son Ödeme (90-120),Son Ödeme (120-150),Son Ödeme (150-180),Son Ödeme (180+)
count,9.176320e+05,9.176320e+05,623908.000000,328193.000000,20902.000000,21664.000000,18893.000000,16995.000000,7323.000000,45708.000000,48281.000000,29005.000000,23030.000000,7.184000e+03,3820.000000,2302.000000,1551.000000,4827.000000
mean,5.010893e+09,5.085794e+02,206.312521,547.264676,643.061367,553.709882,414.438863,536.363810,516.075436,558.643866,583.558553,801.017507,918.617756,1.257710e+03,671.639885,322.164222,393.359884,159.776611
std,2.883691e+09,5.052666e+03,2855.301668,3898.724314,5454.365338,5123.756510,2650.055196,6242.186627,3451.492687,6666.262578,7178.350454,7154.314816,7105.333937,2.007415e+04,5979.226329,3220.519683,5264.099250,2619.161924
min,1.758320e+05,-1.279328e+04,-12793.280000,0.000000,-70.000000,-296.930000,-120.600000,0.000000,-752.000000,-100.660000,-962.000000,0.000000,-6837.000000,-7.010000e+02,-770.000000,-349.000000,0.000000,-15206.880000
25%,2.513721e+09,1.101500e+02,0.320000,116.600000,125.610000,112.922500,119.000000,116.905000,115.000000,121.905000,115.000000,109.000000,77.000000,4.285750e+01,30.330000,24.125000,23.000000,13.470000
50%,5.009565e+09,2.020700e+02,44.220000,210.990000,222.000000,211.000000,203.000000,208.070000,218.540000,213.590000,210.000000,220.880000,172.955000,1.070000e+02,76.000000,55.370000,49.000000,32.000000
75%,7.509497e+09,3.215700e+02,210.220000,334.270000,349.977500,333.000000,320.000000,329.865000,344.985000,330.000000,334.000000,382.470000,320.987500,2.195575e+02,164.000000,124.130000,106.500000,76.000000
max,9.999760e+09,1.173258e+06,799298.890000,429056.570000,393238.000000,319120.000000,152377.000000,560239.000000,158004.450000,550420.870000,550859.750000,694275.910000,221781.470000,1.051591e+06,150233.960000,89285.680000,169410.490000,141070.520000


Tahakkuk - Hamamözü


,sozlesme_hesap_no,kwh
count,1.248180e+05,124818.000000
mean,5.044916e+09,70.874619
std,2.874544e+09,389.217875
min,2.903944e+06,-1242.990000
25%,2.577471e+09,15.490000
50%,5.027442e+09,40.560000
75%,7.594090e+09,70.430000
max,9.991894e+09,25941.600000


Tahakkuk 1 - Gümüşhacıköy


,sozlesme_hesap_no,kwh
count,7.656570e+05,765657.000000
mean,5.019916e+09,97.336632
std,2.881724e+09,1077.758336
min,2.561140e+05,-25370.640000
25%,2.519995e+09,18.570000
50%,5.030422e+09,48.310000
75%,7.517065e+09,82.720000
max,9.998331e+09,153575.730000


Tahakkuk 2 - Göynücek


,sozlesme_hesap_no,kwh
count,2.952230e+05,295223.000000
mean,4.950566e+09,89.669891
std,2.887724e+09,742.276369
min,1.118550e+06,-4208.640000
25%,2.432854e+09,17.860000
50%,4.928206e+09,45.090000
75%,7.475616e+09,77.140000
max,9.997185e+09,105687.690000


---
## 5. İlçelere Göre Benzersiz Müşteri Sayısı

Bu bölümde her ilçedeki benzersiz müşteri sayısı, `sozlesme_hesap_no` sütunu kullanılarak hesaplanmaktadır.

Amaç, üç ilçenin müşteri büyüklüklerini karşılaştırmak ve sonuçları referans değerlerle doğrulamaktır.

In [ ]:
# Her ilçe için benzersiz müşteri sayısını hesaplama

benzersiz_musteriler = {
    "Hamamözü": df_tahakkuk["sozlesme_hesap_no"].nunique(),
    "Gümüşhacıköy": df_tahakkuk_1["sozlesme_hesap_no"].nunique(),
    "Göynücek": df_tahakkuk_2["sozlesme_hesap_no"].nunique()
}

for ilce, musteri_sayisi in benzersiz_musteriler.items():
    print(f"{ilce}: {musteri_sayisi:,}")

Hamamözü: 2,981
Gümüşhacıköy: 18,190
Göynücek: 7,128


### Değerlendirme

Analiz sonucunda en fazla benzersiz müşteriye sahip ilçe **Gümüşhacıköy** (18.190), en az müşteriye sahip ilçe ise **Hamamözü** (2.981) olarak belirlenmiştir. Bu sonuçlar proje dokümanındaki referans değerlerle uyumludur ve veri setinin doğru yüklendiğini doğrulamaktadır.

---
## 6. Tahakkuk Verilerinin Birleştirilmesi

Bu bölümde Hamamözü, Gümüşhacıköy ve Göynücek ilçelerine ait tahakkuk verileri tek bir veri setinde birleştirilmektedir.

Birleştirme işlemi sonrasında toplam kayıt sayısı kontrol edilerek veri bütünlüğü doğrulanacaktır.

In [ ]:
# Üç tahakkuk veri setini birleştirme

df_tahakkuk_tum = pd.concat(
    [df_tahakkuk, df_tahakkuk_1, df_tahakkuk_2],
    ignore_index=True
)

print("Toplam kayıt sayısı:", len(df_tahakkuk_tum))
print("Veri boyutu:", df_tahakkuk_tum.shape)

Toplam kayıt sayısı: 1185698
Veri boyutu: (1185698, 10)


---
## 7. Eksik Değer Analizi

Bu bölümde birleştirilen tahakkuk veri setindeki eksik değerler incelenmektedir.

Her sütun için eksik kayıt sayısı ve eksik veri oranı hesaplanarak veri kalitesi değerlendirilmektedir.

In [11]:
# Eksik değer analizi

eksik_degerler = pd.DataFrame({
    "Eksik Değer Sayısı": df_tahakkuk_tum.isnull().sum(),
    "Eksik Değer Oranı (%)": (
        df_tahakkuk_tum.isnull().mean() * 100
    ).round(2)
})

eksik_degerler

,Eksik Değer Sayısı,Eksik Değer Oranı (%)
il,0,0.0
ilce,0,0.0
sozlesme_hesap_no,0,0.0
mali_yil_donem,0,0.0
fatura_tarihi,0,0.0
kayit_tarihi,0,0.0
vade_tarihi,0,0.0
hesap_sinifi,0,0.0
Hesap Sınıfı,0,0.0
kwh,0,0.0


### Değerlendirme

Birleştirilen tahakkuk veri setinde yapılan inceleme sonucunda hiçbir sütunda eksik değere rastlanmamıştır.

Bu durum veri setinin analiz için uygun olduğunu ve veri bütünlüğünün korunduğunu göstermektedir.

---
## 8. Negatif kWh Değerlerinin İncelenmesi

Bu bölümde `kwh` sütununda bulunan negatif tüketim değerleri tespit edilmektedir.

Negatif değerler doğrudan silinmeden önce sayıları ve örnek kayıtları incelenecektir.

In [ ]:
# Negatif kWh değerlerini filtreleme

negatif_kwh = df_tahakkuk_tum[df_tahakkuk_tum["kwh"] < 0]

print("Negatif kWh kayıt sayısı:", len(negatif_kwh))
print("Negatif kWh oranı (%):", round(len(negatif_kwh) / len(df_tahakkuk_tum) * 100, 4))

display(negatif_kwh.head())

Negatif kWh kayıt sayısı: 151
Negatif kWh oranı (%): 0.0127


,il,ilce,sozlesme_hesap_no,mali_yil_donem,fatura_tarihi,kayit_tarihi,vade_tarihi,hesap_sinifi,Hesap Sınıfı,kwh
23573,AMASYA,HAMAMÖZÜ,7904741733,2025-03-01,2025-04-14,2025-06-10,2025-04-22,M001,Mesken,-1242.99
23576,AMASYA,HAMAMÖZÜ,7904741733,2025-04-01,2025-04-14,2025-06-10,2025-04-22,M001,Mesken,-733.04
28413,AMASYA,HAMAMÖZÜ,469034964,2023-10-01,2023-10-11,2023-11-28,2023-10-23,M001,Mesken,-1.04
70626,AMASYA,HAMAMÖZÜ,7722066065,2023-10-01,2023-10-23,2023-11-28,2023-11-02,M001,Mesken,-1.34
180501,AMASYA,GÜMÜŞHACIKÖY,2945215788,2024-02-01,2024-04-15,2024-06-15,2024-03-25,M001,Mesken,-374.09


### Değerlendirme

Analiz sonucunda **151 adet negatif kWh değeri** tespit edilmiştir. Bu kayıtlar toplam veri setinin yaklaşık **%0,013'ünü** oluşturmaktadır.

Negatif tüketim değerleri doğrudan hatalı veri olarak değerlendirilmemiştir. Negatif tüketim kayıtları sayaç düzeltmeleri, mahsuplaşmalar, fatura iptalleri veya veri giriş süreçleriyle ilişkili olabilir. Ancak mevcut veri seti tek başına bu kayıtların kesin nedenini açıklamamaktadır.

Bu nedenle bu aşamada herhangi bir kayıt silinmemiş, yalnızca tespit edilerek raporlanmıştır.

---
## 9. Aykırı Değer (Outlier) Analizi

Bu bölümde `kwh` sütunundaki aykırı değerler IQR (Interquartile Range) yöntemi kullanılarak tespit edilmektedir.

Aykırı değerler analiz edilmek amacıyla belirlenmiş olup bu aşamada veri setinden çıkarılmamıştır.

In [13]:
# IQR yöntemi ile aykırı değer analizi

Q1 = df_tahakkuk_tum["kwh"].quantile(0.25)
Q3 = df_tahakkuk_tum["kwh"].quantile(0.75)

IQR = Q3 - Q1

alt_sinir = Q1 - 1.5 * IQR
ust_sinir = Q3 + 1.5 * IQR

aykiri_degerler = df_tahakkuk_tum[
    (df_tahakkuk_tum["kwh"] < alt_sinir) |
    (df_tahakkuk_tum["kwh"] > ust_sinir)
]

print(f"Alt sınır : {alt_sinir:.2f}")
print(f"Üst sınır : {ust_sinir:.2f}")
print(f"Aykırı değer sayısı : {len(aykiri_degerler)}")
print(f"Aykırı değer oranı : %{len(aykiri_degerler) / len(df_tahakkuk_tum) * 100:.2f}")

display(aykiri_degerler.head())

Alt sınır : -74.97
Üst sınır : 172.98
Aykırı değer sayısı : 48554
Aykırı değer oranı : %4.09


,il,ilce,sozlesme_hesap_no,mali_yil_donem,fatura_tarihi,kayit_tarihi,vade_tarihi,hesap_sinifi,Hesap Sınıfı,kwh
346,AMASYA,HAMAMÖZÜ,3154984217,2023-02-01,2023-03-10,2023-05-11,2023-03-20,M001,Mesken,175.30
348,AMASYA,HAMAMÖZÜ,3154984217,2023-03-01,2023-04-10,2023-05-24,2023-04-24,M001,Mesken,189.22
354,AMASYA,HAMAMÖZÜ,3154984217,2023-06-01,2023-07-12,2023-09-07,2023-07-24,M001,Mesken,228.95
356,AMASYA,HAMAMÖZÜ,3154984217,2023-07-01,2023-08-11,2024-01-03,2023-08-21,M001,Mesken,223.68
358,AMASYA,HAMAMÖZÜ,3154984217,2023-08-01,2023-09-12,2023-10-31,2023-09-22,M001,Mesken,212.69


### Değerlendirme

IQR yöntemi sonucunda **48.554 adet aykırı değer** tespit edilmiştir. Bu kayıtlar toplam veri setinin yaklaşık **%4,09'unu** oluşturmaktadır.

Aykırı olarak belirlenen değerler istatistiksel yönteme göre hesaplanmıştır. Ancak enerji tüketim verilerinde yüksek tüketimli aboneler doğal olarak bu aralığın dışında kalabilmektedir.

Bu nedenle aykırı değerler veri setinden çıkarılmamış, yalnızca analiz kapsamında raporlanmıştır.

---
## 10. Hesap Sınıfına Göre Elektrik Tüketim Analizi

Bu bölümde hesap sınıflarına göre elektrik tüketim değerlerinin temel istatistikleri hesaplanmaktadır.

Her hesap sınıfı için;

- Kayıt sayısı
- Ortalama tüketim (kWh)
- Medyan tüketim (kWh)
- Standart sapma

değerleri hesaplanarak tüketim davranışları karşılaştırılmaktadır.

In [14]:
# Hesap sınıfına göre temel istatistikler

hesap_sinifi_istatistik = (
    df_tahakkuk_tum
    .groupby("Hesap Sınıfı")["kwh"]
    .agg(
        Kayit_Sayisi="count",
        Ortalama_kWh="mean",
        Medyan_kWh="median",
        Standart_Sapma="std"
    )
    .round(2)
    .sort_values("Ortalama_kWh", ascending=False)
)

hesap_sinifi_istatistik

,Kayit_Sayisi,Ortalama_kWh,Medyan_kWh,Standart_Sapma
Hesap Sınıfı,,,,
Karayolları Genel Müdürlüğü Aydınlatma,41,30203.43,36810.90,16963.57
Aritma Tesisleri,35,16594.17,16186.91,11656.71
Lisansız Üreticiler,180,16155.25,322.32,35926.42
Sanayi,187,7293.80,1531.68,13716.34
İçme-Kullanma Suyu (Belediye),417,5213.51,400.02,8948.07
...,...,...,...,...
"Bina Ort Kul (Asn,Hidr,Kapıcı Dai vb.)",3505,29.97,15.76,41.33
"Resmi Hayır Kurumları,Dernek,Vakıf,Müze",169,23.00,8.82,42.50
Cemevleri,916,10.71,5.62,16.45


### Değerlendirme

Hesap sınıflarına göre yapılan analiz sonucunda en yüksek ortalama elektrik tüketiminin **Karayolları Genel Müdürlüğü Aydınlatma**, **Arıtma Tesisleri**, **Lisanssız Üreticiler** ve **Sanayi** abonelerinde olduğu görülmüştür.

Mesken ve benzeri bireysel abonelerde ortalama tüketim daha düşük seviyelerde gerçekleşirken, kurumsal ve endüstriyel abonelerde tüketim değerleri belirgin şekilde yüksektir. Bu sonuçlar hesap sınıfları arasındaki tüketim farklılıklarını açıkça göstermektedir.

Ayrıca IQR sınırları tüm hesap sınıfları birlikte değerlendirilerek hesaplanmıştır. Hesap sınıflarının tüketim yapıları farklı olduğundan, ileri analizlerde aykırı değerlerin hesap sınıfı bazında ayrıca incelenmesi daha sağlıklı olacaktır.

---
# Notebook 01 Sonuç

Bu notebook kapsamında enerji perakende veri setleri başarıyla yüklenmiş ve ön analiz çalışmaları tamamlanmıştır.

Gerçekleştirilen çalışmalar:

- Veri setleri başarıyla yüklenmiş ve doğrulanmıştır.
- Veri yapısı incelenmiştir.
- İlçelere göre benzersiz müşteri sayıları hesaplanmıştır.
- Tahakkuk verileri tek bir veri setinde birleştirilmiştir.
- Eksik değer, negatif tüketim ve aykırı değer analizleri gerçekleştirilmiştir.
- Hesap sınıflarına göre temel tüketim istatistikleri hesaplanmıştır.

Bu analizler sonucunda veri setinin görselleştirme ve ileri analizler için kullanılabilir olduğu değerlendirilmiştir. Ancak negatif ve aykırı kWh kayıtlarının sonraki analizlerde dikkate alınması gerekmektedir.

Bir sonraki notebook'ta elde edilen bulgular grafikler ve görsel analizlerle desteklenecektir.
